src/v/cluster/types.cc

```diff
@@ -1370,4 +1370,19 @@ adl<cluster::set_maintenance_mode_reply>::from(iobuf_parser& parser) {
     return cluster::set_maintenance_mode_reply{.error = error};
 }
 
+void adl<cluster::partition_assignment>::to(
+  iobuf& out, cluster::partition_assignment&& p_as) {
+    reflection::serialize(out, p_as.group, p_as.id, std::move(p_as.replicas));
+}
+
+cluster::partition_assignment
+adl<cluster::partition_assignment>::from(iobuf_parser& parser) {
+    return cluster::partition_assignment{
+      .group = reflection::adl<raft::group_id>{}.from(parser),
+      .id = reflection::adl<model::partition_id>{}.from(parser),
+      .replicas = reflection::adl<std::vector<model::broker_shard>>{}.from(
+        parser),
+    };
```

Comment:
does the use of designated initializers here mean that the type partition_assignment is still an aggregate type? and if so, does that mean that reflection is still available to the type? it might make sense to add a user-defined constructor in this case so it isn't an aggregate and won't ever be accidentally used with reflection.